webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/ProjectExportService.scala

```diff
@@ -160,33 +176,83 @@ final case class ProjectExportServiceLive(
     val graphIri = permissionsDataNamedGraph
     val file     = NamedGraphTrigFile(graphIri, tempDir)
     for {
-      query <- ZIO.attempt(twirl.queries.sparql.admin.txt.getProjectPermissions(project.id.value))
+      query <- ZIO.attempt(getProjectPermissions(project.id.value))
       _     <- triplestoreService.sparqlHttpConstructFile(query.toString(), graphIri, file.dataFile, TriG)
     } yield file
   }
 
   private def mergeDataToFile(allData: Seq[NamedGraphTrigFile], targetFile: Path): Task[Path] =
     TriGCombiner.combineTrigFiles(allData.map(_.dataFile), targetFile)
 
-  override def exportProject(project: KnoraProject): Task[Path] = ZIO.scoped {
+  override def exportProject(project: KnoraProject, user: UserADM): Task[Path] = ZIO.scoped {
     for {
-      exportDir  <- Files.createTempDirectory(Some(s"export-${project.shortname}"), fileAttributes = Nil)
-      collectDir <- Files.createTempDirectoryScoped(Some(project.shortname), fileAttributes = Nil)
-      _          <- exportProjectTriples(project, trigExportFile(project, collectDir))
-      _          <- exportProjectAssets(project, collectDir)
-      zipped     <- ZipUtility.zipFolder(collectDir, exportDir)
+      exportDir    <- Files.createTempDirectory(Some(s"export-${project.shortname}"), fileAttributes = Nil)
+      collectDir   <- Files.createTempDirectoryScoped(Some(project.shortname), fileAttributes = Nil)
+      _            <- exportProjectTriples(project, trigExportFile(project, collectDir))
+      _            <- exportProjectAssets(project, collectDir, user)
+      zipped       <- ZipUtility.zipFolder(collectDir, exportDir)
+      fileSize     <- Files.size(zipped)
+      absolutePath <- zipped.toAbsolutePath
+      _            <- ZIO.logInfo(s"Exported project ${project.shortname} to $absolutePath ($fileSize bytes)")
     } yield zipped
   }
 
-  private def exportProjectAssets(project: KnoraProject, tempDir: Path) = {
+  private def exportProjectAssets(project: KnoraProject, tempDir: Path, user: UserADM): ZIO[Any, Throwable, Path] = {
     val exportedAssetsDir = tempDir / "assets"
     for {
       _ <- Files.createDirectory(exportedAssetsDir)
+      _ <- assetService.exportProjectAssets(project, exportedAssetsDir, user)
     } yield exportedAssetsDir
   }
 }
 
+trait AssetService {
+  def exportProjectAssets(project: KnoraProject, tempDir: Path, user: UserADM): Task[Path]
+}
+case class Asset(belongsToProject: KnoraProject, internalFilename: String)
+object Asset {
+  def logString(asset: Asset) = s"asset:${asset.belongsToProject.shortcode}/${asset.internalFilename}"
```

Comment:
would it make sense here to use the actual path, as SIPI uses it now (with the two subfolders according to the file name), or is this something different entirely?

Reply:
It's just a log string so in all logs it is the same also `shortcode/filename` is actually `prefix/identifier` in Sipi.